client/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces.ts

```diff
@@ -0,0 +1,71 @@
+import { ReactNode } from 'react';
+
+export interface ItemsDataViewsType< T > {
+	items: T[] | undefined;
+	pagination: DataViewsPaginationInfo;
+	searchLabel?: string;
+	fields: DataViewsColumn[];
+	actions?: DataViewsAction[];
+	getItemId?: ( item: T ) => string;
+	itemFieldId?: string; // The field path to get the item id. Examples `id` or `site.blog_id`
+	setDataViewsState: ( callback: ( prevState: DataViewsState ) => DataViewsState ) => void;
+	dataViewsState: DataViewsState;
+}
+
```

Comment:
We have added new interfaces for each DataViews element: Column, Action, PaginationInfo, Sort, Filter.